### PR TITLE
Estimate width for ASK_14

### PR DESCRIPTION
**Pull request by SnowNooDLe**
# Estimate width for ASK_14

This estimation is model specified and we don't `width` to be everywhere, hence, we decided to put this into the wrapper instead of the `new_calc_emp_ds` script.

The equation was found from the NGA-West2 spreadsheet(Thanks to Robin)
Unfortunately, this equation was not mentioned in the paper, hence, we cannot justify implementing it within the OpenQuake but we believe this equation is okay enough to use.

### CODE REVIEW

File: empirical/util/openquake_wrapper_vectorized.py [hunk 1]
```diff
@@ -155,15 +155,15 @@ def interpolate_with_pga(
 
 
 def oq_run(
-    model: Enum,
+    model_enum: Enum,
```
Comment (jasonmotha) [inline empirical/util/openquake_wrapper_vectorized.py:158]: Why did you rename this variable?

It makes me think it's the name of the enum rather than a value of one of the entries in the enum.

Comment (SnowNooDLe) [reply empirical/util/openquake_wrapper_vectorized.py:158]: Otherwise, the `model` is overwritten by line 183.
So I had to either rename `model` or bring the estimation before the model instantiation, and I preferred the first option :)

Comment (jasonmotha) [reply empirical/util/openquake_wrapper_vectorized.py:158]: I'd call this one `model_name`

And the other one

`model_function` maybe?

Comment (claudio525) [reply empirical/util/openquake_wrapper_vectorized.py:158]: What about `model_type` instead of `model_enum`?

Comment (SnowNooDLe) [reply empirical/util/openquake_wrapper_vectorized.py:158]: How about `GMM`?
man, naming is hard.

**Review by jasonmotha — APPROVED** [review]

File: empirical/util/openquake_wrapper_vectorized.py (hunk 1 above)
Comment (SnowNooDLe) [reply empirical/util/openquake_wrapper_vectorized.py]: I think `model_type` seems more fluent by those reasons
- `OQ_MODELS[model_type][tect_type]` : both type type from OQ_MODELS dictionary
- model_type.name.endswith("_NZ"): `model_name.name` looks a bit odd

File: empirical/util/openquake_wrapper_vectorized.py [hunk 2]
```diff
@@ -204,6 +204,13 @@ def oq_run(
     # Make a copy in case the original rupture_df used with other functions
     rupture_df = rupture_df.copy()
 
+    # Model specified estimation that cannot be done within OQ as paper does not specify
+    if model_type.name == "ASK_14" and "width" not in rupture_df:
+        rupture_df["width"] = np.minimum(
```
Comment (claudio525) [inline empirical/util/openquake_wrapper_vectorized.py:209]: I think it would makes sense to make this a separate function and have it with all the other estimation functions?

Comment (jasonmotha) [reply empirical/util/openquake_wrapper_vectorized.py:209]: This width function is ASK specific so its utility should be reserved for this function only. 
I don't mind it in a function but it shouldn't be used elsewhere.

File: empirical/util/estimations.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import numpy as np
+import pandas as pd
+
+
+def estimate_width_ASK14(rupture_df: pd.DataFrame):
```
Comment (jasonmotha) [inline empirical/util/estimations.py:5]: Ah, whoops. Ideally this takes in two arrays rather than the whole dataframe and the oq_wrapper deals with the dataframe.

**Review by jasonmotha — COMMENTED** [review]
One quick comment. Otherwise looks good.

File: empirical/util/estimations.py (hunk 3 above)
Comment (SnowNooDLe) [reply empirical/util/estimations.py:5]: Ahhhhh so something like

```python
    if model_type.name == "ASK_14" and "width" not in rupture_df:
        rupture_df["width"] = estimations.estimate_width_ASK14(rupture_df["dip"], rupture_df["mag"])
```

Comment (jasonmotha) [reply empirical/util/estimations.py:5]: Yep.

**Review by jasonmotha — APPROVED** [review]

**Review by claudio525 — APPROVED** [review]
